src/main/scala/org/holmesprocessing/totem/services/cfgangr/convertbinary.py

```diff
@@ -1,21 +1,37 @@
 import angr
-
+import simuvex
+import cle
+import os
+import json
 # imports for graph handling
 import networkx as nx
 from networkx.readwrite import json_graph
 
 
-def generateCFG(binary, analysisType = 'Fast'):
-    # Create the angr project for the binary
-    project = angr.Project(binary, auto_load_libs = False)
+def generateCFG(binary, max_size, analysisType = 'Fast'):
+    response = {}
+    try:
+        # Create the angr project for the binary
+        project = angr.Project(binary, auto_load_libs=False)
+        binary_size = os.stat(binary).st_size
+        max_binary_size = int(max_size) * 1024  # Here we will set a limit of binary files in bytes
+        if binary_size > max_binary_size:
+            response['error'] = 'CFG generation failed because the binary size is too big'
```

Comment:
We should have these raised and then have the main program return an appropriate error code. The error codes are not finalized or complete .... but for reference:

https://docs.holmesprocessing.com/en/latest/rst/components/holmes-totem/extending/services/writing_a_service/index.html#http-error-codes